### PR TITLE
DUP: move AMQP handling logic to Mississippi

### PR DESCRIPTION
**Pull request by Annopaolo**
* [x] I have read [CONTRIBUTING.md](../CONTRIBUTING.md) and [CODE_OF_CONDUCT.md](../CODE_OF_CONDUCT.md)
* [x] I have added to [CHANGELOG.md](../CHANGELOG.md) relevant changes and any other user facing change
* [x] I have added or ran the appropriate tests

#### What this PR does / why we need it:
Use the Mississippi framework to manage messages coming from VerneMQ. This allows to retain the same ordering and delivery guarantees, while moving to the external component all the intricacies of handling messages in a distributed setting.
This will also increase the ease of contributing to the core message handling logic, as now code only works on what to do once a message is arrived, instead of also including how to receive a message.

Note that e2e tests will fail until https://github.com/astarte-platform/astarte_vmq_plugin/pull/94 is merged.

#### Which issue(s) this PR fixes:
(none)

##### Does this PR introduce a user-facing change?
* [ ] Yes
* [x] No

#### Additional documentation e.g. usage docs, diagrams, etc.:

```docs

```

### CODE REVIEW

File: apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/volatile_trigger_handler.ex [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  def install_volatile_trigger(%InstallVolatileTrigger{
+        realm_name: realm_name,
+        device_id: encoded_device_id,
+        object_id: object_id,
+        object_type: object_type,
+        parent_id: parent_id,
+        simple_trigger_id: simple_trigger_id,
+        simple_trigger: simple_trigger,
+        trigger_target: trigger_target
+      }) do
```
Comment (noaccOS) [inline apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/volatile_trigger_handler.ex:37]:
```suggestion
  def install_volatile_trigger(install_volatile_trigger) do
    %InstallVolatileTrigger{
        realm_name: realm_name,
        device_id: encoded_device_id,
        object_id: object_id,
        object_type: object_type,
        parent_id: parent_id,
        simple_trigger_id: simple_trigger_id,
        simple_trigger: simple_trigger,
        trigger_target: trigger_target
      } = install_volatile_trigger
```

I'd suggest keeping the pattern matching complexity out of the function definition when possible (ie when we're not pattern matching on the function arguments)

File: apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/volatile_trigger_handler.ex [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    with :ok <- verify_device_exists(realm_name, encoded_device_id),
+         {:ok, device_id} <- decode_device_id(encoded_device_id) do
+      sharding_key = {realm_name, device_id}
+
+      {:ok, du_pid} = DataUpdater.get_data_updater_process(sharding_key)
+
+      DataUpdater.handle_signal(
+        du_pid,
+        {:handle_install_volatile_trigger, object_id, object_type, parent_id, simple_trigger_id,
+         simple_trigger, trigger_target}
```
Comment (noaccOS) [inline apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/volatile_trigger_handler.ex:47]: why not passing down the whole object? as a rule of thumb, when you have more than 4-5 distinct objects you're usually better off aggregating them in a single collection (keyword, struct, etc)

File: apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex [hunk 3]
```diff
@@ -87,15 +97,78 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
...
+      _ ->
+        # ack to discard unhandled messages
+        {:ack, :ok, state}
+    end
+  end
+
+  @impl Handler
+  def handle_signal(signal, state) do
+    case signal do
+      {:handle_install_volatile_trigger, object_id, object_type, parent_id, simple_trigger_id,
```
Comment (noaccOS) [inline apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex:154]: this is the only `handle_` function where we pattern match on the argument inside the function, I'd keep this at function signature level for consistency but don't feel strongly about it

**Review by noaccOS — COMMENTED** [review]
I see you have split the PR in many commits, but I think that
1. They don't really work by themselves because they remove functionality in order to implement them one at a time
2. They don't make the review process any easier because the commit diffs are gigantic with all the comment - uncomment procedure

Still, aside from a couple comments, the code looks good

File: apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/application.ex [hunk 4]
```diff
@@ -63,15 +64,16 @@ defmodule Astarte.DataUpdaterPlant.Application do
...
       amqp_consumer_options: [host: Config.amqp_consumer_host!()],
-      events_consumer_connection_number: Config.amqp_consumer_connection_number!(),
-      mississippi_queues_config: [
-        events_exchange_name: Config.events_exchange_name!(),
-        data_queue_prefix: Config.data_queue_prefix!(),
-        data_queue_range_start: Config.data_queue_range_start!(),
-        data_queue_range_end: Config.data_queue_range_end!(),
-        data_queue_total_count: Config.data_queue_total_count!()
-      ],
-      message_handler: Impl
```
Comment (noaccOS) [inline apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/application.ex]: Should this change be in 209fa627?

Comment (Annopaolo) [reply apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/application.ex]: Makes sense

**Review by noaccOS — APPROVED** [review]

**Review by rbino — APPROVED** [review]

**Pull request merged**